README.md

```diff
@@ -144,33 +144,105 @@ $ make -C build.assets build-binaries
 
 ### Local Build
 
-To perform a build on your host, ensure you have installed Go. In order to
-include the Rust-powered features like Desktop Access, you'll
-also need `cargo` and `rustc`. The current versions of these tools can be found
-in `build.assets/Makefile`.
+**Important:**
 
-```bash
-# get the source & build:
+* The Go compiler is somewhat sensitive to the amount of memory: you will need
+  **at least** 1GB of virtual memory to compile Teleport. A 512MB instance
+  without swap will **not** work.
+* This will build the latest version of Teleport, **regardless** of whether it
+  is stable. If you want to build the latest stable release, run `git checkout`
+  to the corresponding tag (for example, run `git checkout v8.0.0`) **before**
+  performing a build.
+* The following instructions are Mac-specific.
+
+#### Dependencies
+
+Ensure you have installed correct versions of dependencies:
+
+* `Go` version from [go.mod](https://github.com/gravitational/teleport/blob/0e9d621b464f2bd05a28e7db0ae2ef65349c460e/go.mod#L3)
+  
+  ```shell
+  # if we are not on the latest, you might need to
+  # brew install go@<some_version>, i.e. 1.16
+  #
+  # check which version will be installed by running
+  # brew info go
+  
+  $ brew install go
+  ````
+  
+If you wish to build the Rust-powered features like Desktop Access:
+
+* `Rust` and `Cargo` version from [build.assets/Makefile](https://github.com/gravitational/teleport/blob/c3dde989cc9de6fa199f1a12287926218cc2e685/build.assets/Makefile#L21):
```

Comment:
This one is tougher to link without fixing the commit, but maybe:

```suggestion
* `Rust` and `Cargo` version from
  [build.assets/Makefile](https://github.com/gravitational/teleport/blob/master/build.assets/Makefile#L21)
  (search for RUST_VERSION):
```

Reply:
I'll fix it